### PR TITLE
Add hessian plugin tests cases.

### PR DESCRIPTION
**Pull request by wayilau**
Please fill up those following information before send Pull request 

* Pull request NO: 2644
* Agent repository URL: https://github.com/wayilau/skywalking
* Testcases repository URL: http://localhost:{SERVER_OUTPUT_PORT}/hello

### CODE REVIEW

File: hessian-scenario/src/site/fr/apt/format.apt [hunk 1]
```diff
@@ -0,0 +1,602 @@
+ -----
```
Comment (wu-sheng) [inline hessian-scenario/src/site/fr/apt/format.apt]: What is this file?

File: hessian-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      /HelloService,
+      /hello
+    ]
+  }
+  ]
+},
+"segmentItems: [
+{
+  applicationCode: hessian-consumer,
+  segmentSize: 1,
```
Comment (wu-sheng) [inline hessian-scenario/config/expectedData.yaml]: Why only one segment? And where is the segment ref?

**Review by wu-sheng — CHANGES_REQUESTED** [review]
Complete the tests to match A -> B RPC scenario. Also, provide test report at your local env

**Comment by wayilau** [comment]
i will provide my local test reports.

File: hessian-scenario/src/site/fr/apt/format.apt (hunk 1 above)
Comment (wayilau) [reply hessian-scenario/src/site/fr/apt/format.apt]: deleted.

File: hessian-scenario/config/expectedData.yaml (hunk 2 above)
Comment (wayilau) [reply hessian-scenario/config/expectedData.yaml]: resolving.

**Comment by wayilau** [comment]
@wu-sheng why my tests success when deal with seperate version, but failed with all version together. Also my actual data may lost some data when all version tests together. Is any problem with the tests tool? or my code problem.  I will show you the tests results log.

**Comment by wayilau** [comment]
{
  "registryItems": {
    "applications": [
      {
        "hessian-consumer": "nq 0"
      },
      {
        "hessian-provider": "nq 0"
      }
    ],
    "instances": [
      {
        "hessian-consumer": 1
      },
      {
        "hessian-provider": 1
      }
    ],
    "operationNames": [
      {
        "hessian-consumer": [
          "/HelloService",
          "/hessian-case/hello"
        ]
      },
      {
        "hessian-provider": [
          "org.apache.skywalking.plugin.hessian.service.HelloService"
        ]
      }
    ]
  },
  "segmentItems": [
    {
      "applicationCode": "hessian-consumer",
      "segmentSize": "1",
      "segments": [
        {
          "segmentId": "not null",
          "spans": [
            {
              "operationName": "/HelloService",
              "operationId": "eq 0",
              "parentSpanId": "0",
              "spanId": "1",
              "spanLayer": "RPCFramework",
              "tags": [
                {
                  "key": "url",
                  "value": "/HelloService"
                }
              ],
              "logs": [],
              "startTime": "nq 0",
              "endTime": "nq 0",
              "componentId": "65",
              "isError": false,
              "spanType": "Exit",
              "peerId": "eq 0"
            },
            {
              "operationName": "/hessian-case/hello",
              "operationId": "eq 0",
              "parentSpanId": "-1",
              "spanId": "0",
              "spanLayer": "Http",
              "tags": [
                {
                  "key": "url",
                  "value": "http://localhost:18080/hessian-case/hello"
                },
                {
                  "key": "http.method",
                  "value": "GET"
                }
              ],
              "logs": [],
              "startTime": "nq 0",
              "endTime": "nq 0",
              "componentId": "1",
              "isError": false,
              "spanType": "Entry",
              "peerId": "eq 0"
            }
          ]
        }
      ]
    },
    {
      "applicationCode": "hessian-provider",
      "segmentSize": "1",
      "segments": [
        {
          "segmentId": "not null",
          "spans": [
            {
              "operationName": "org.apache.skywalking.plugin.hessian.service.HelloService",
              "operationId": "eq 0",
              "parentSpanId": "-1",
              "spanId": "0",
              "spanLayer": "RPCFramework",
              "tags": [
                {
                  "key": "url",
                  "value": "org.apache.skywalking.plugin.hessian.service.HelloService"
                }
              ],
              "refs": [
                {
                  "parentEndpointId": 0,
                  "parentEndpoint": "/hessian-case/hello",
                  "networkAddressId": 0,
                  "entryEndpointId": 0,
                  "refType": "CrossProcess",
                  "parentSpanId": 1,
                  "parentTraceSegmentId": "not null",
                  "parentServiceInstanceId": "nq 0",
                  "networkAddress": ":",
                  "entryEndpoint": "/hessian-case/hello",
                  "entryServiceInstanceId": "nq 0"
                }
              ],
              "startTime": "not null",
              "endTime": "not null",
              "componentId": "65",
              "isError": false,
              "spanType": "Entry",
              "peerId": "eq 0"
            }
          ]
        }
      ]
    }
  ]
}
[2019-06-18 11:06:10:358] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.DataAssert.assertEquals(DataAssert.java:13) - actual data:
{
  "registryItems": {
    "applications": [
      {
        "hessian-consumer": 2
      },
      {
        "hessian-provider": 3
      }
    ],
    "instances": [
      {
        "hessian-consumer": 1
      },
      {
        "hessian-provider": 1
      }
    ],
    "operationNames": [
      {
        "hessian-consumer": [
          "/HelloService",
          "/hessian-case/hello"
        ]
      },
      {
        "hessian-provider": [
          "org.apache.skywalking.plugin.hessian.service.HelloService"
        ]
      }
    ]
  },
  "segmentItems": [
    {
      "applicationCode": "hessian-consumer",
      "segmentSize": "1",
      "segments": [
        {
          "segmentId": "2.26.15608271058940000",
          "spans": [
            {
              "operationName": "/HelloService",
              "operationId": "0",
              "parentSpanId": "0",
              "spanId": "1",
              "spanLayer": "RPCFramework",
              "tags": [
                {
                  "key": "url",
                  "value": "/HelloService"
                }
              ],
              "startTime": "1560827106139",
              "endTime": "1560827106645",
              "componentId": "65",
              "componentName": "",
              "isError": false,
              "spanType": "Exit",
              "peer": ":",
              "peerId": "0"
            },
            {
              "operationName": "/hessian-case/hello",
              "operationId": "0",
              "parentSpanId": "-1",
              "spanId": "0",
              "spanLayer": "Http",
              "tags": [
                {
                  "key": "url",
                  "value": "http://localhost:18180/hessian-case/hello"
                },
                {
                  "key": "http.method",
                  "value": "GET"
                }
              ],
              "startTime": "1560827105903",
              "endTime": "1560827106723",
              "componentId": "1",
              "componentName": "",
              "isError": false,
              "spanType": "Entry",
              "peer": "",
              "peerId": "0"
            }
          ]
        }
      ]
    },
    {
      "applicationCode": "hessian-provider",
      "segmentSize": "1",
      "segments": [
        {
          "segmentId": "1.26.15608271063980000",
          "spans": [
            {
              "operationName": "org.apache.skywalking.plugin.hessian.service.HelloService",
              "operationId": "0",
              "parentSpanId": "-1",
              "spanId": "0",
              "spanLayer": "RPCFramework",
              "tags": [
                {
                  "key": "url",
                  "value": "org.apache.skywalking.plugin.hessian.service.HelloService"
                }
              ],
              "refs": [
                {
                  "parentEndpointId": 0,
                  "parentEndpoint": "/hessian-case/hello",
                  "networkAddressId": 0,
                  "entryEndpointId": 0,
                  "refType": "CrossProcess",
                  "parentSpanId": 1,
                  "parentTraceSegmentId": "2.26.15608271058940000",
                  "parentServiceInstanceId": 2,
                  "networkAddress": ":",
                  "entryEndpoint": "/hessian-case/hello",
                  "entryServiceInstanceId": 2
                }
              ],
              "startTime": "1560827106583",
              "endTime": "1560827106657",
              "componentId": "65",
              "componentName": "",
              "isError": false,
              "spanType": "Entry",
              "peer": "",
              "peerId": "0"
            }
          ]
        }
      ]
    }
  ]
}
[2019-06-18 11:06:10:364] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.RegistryItemsAssert.assertEquals(RegistryItemsAssert.java:12) - registry applications assert successful.
[2019-06-18 11:06:10:372] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.RegistryItemsAssert.assertEquals(RegistryItemsAssert.java:14) - registry instances assert successful.
[2019-06-18 11:06:10:373] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.RegistryItemsAssert.assertEquals(RegistryItemsAssert.java:16) - registry operation name assert successful.
[2019-06-18 11:06:10:379] [ERROR] - org.skywalking.apm.test.agent.tool.Main.main(Main.java:41) -
assert failed.
SegmentNotFoundException:
expected:
  Segment:
  - span[0, 1] /HelloService
  - span[-1, 0] /hessian-case/hello

actual:
  Segment[2.26.15608271058940000] e
  expected:     Span[-1, 0] /hessian-case/hello
  actual:       span[-1, 0] /hessian-case/hello
  reason:       [tag(url) value]: expected=>{}, actual=>{http://localhost:18080/hessian-case/hello}

**Comment by wu-sheng** [comment]
@ascrutae @IanCao Please provide some helps.

**Comment by wayilau** [comment]
@wu-sheng  test_report:
# Test Report - 2019-06-21 18:04

- tests  : **2 passed**. **4 failed**
- branch name : **[feature2](https://github.com/apache/incubator-skywalking/tree/feature2)**
- commit id : **[baf4c465abd141a7cf2b51a5dac3c8a6a14ea108](https://github.com/apache/incubator-skywalking/commit/baf4c465abd141a7cf2b51a5dac3c8a6a14ea108)**
- cases branch: **[master](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/tree/master)**
- cases commit id: **[cb12e61b06dc4eb8f3c28756adff1a87741e41fe](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/commit/cb12e61b06dc4eb8f3c28756adff1a87741e41fe)**

## Cases List

| Case description | Status | Cases|
|:-----|:-----:|:-----:|
|HESSIAN| **2 passed. 4 failed**| [click me](#hessian) |

## HESSIAN

###
|  Version     | Passed | Failed|
|:------------- |:-------:|:-----:|
| 4.0.60  | |:heavy_check_mark:|
| 4.0.51  | |:heavy_check_mark:|
| 4.0.59  | :heavy_check_mark:||
| 4.0.52  | :heavy_check_mark:||
| 4.0.53  | |:heavy_check_mark:|
| 4.0.38  | |:heavy_check_mark:|

**Comment by wayilau** [comment]
The report is test all versions together. actually they are successful when test with seperate version. But here only two success.

**Comment by wu-sheng** [comment]
> The report is test all versions together. actually they are successful when test with seperate version. But here only two success.

Is this fixed today?

**Comment by wayilau** [comment]
> > The report is test all versions together. actually they are successful when test with seperate version. But here only two success.
> 
> Is this fixed today?

May be i need some help. @wu-sheng

**Comment by wu-sheng** [comment]
@IanCao Please take a look

**Comment by wu-sheng** [comment]
@IanCao @ascrutae Please run the test case, let's see the result.

**Comment by wayilau** [comment]
# Test Report - 2019-07-04 16:28

- tests  : **3 passed**. **3 failed**
- branch name : **[feature2](https://github.com/apache/incubator-skywalking/tree/feature2)**
- commit id : **[e81507d73a9783dd0af8ebb7d2065c77c790903b](https://github.com/apache/incubator-skywalking/commit/e81507d73a9783dd0af8ebb7d2065c77c790903b)**
- cases branch: **[master](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/tree/master)**
- cases commit id: **[e45d662423572fd9761cbb106aed5c7df6240ccc](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/commit/e45d662423572fd9761cbb106aed5c7df6240ccc)**

## Cases List

| Case description | Status | Cases|
|:-----|:-----:|:-----:|
|HESSIAN| **3 passed. 3 failed**| [click me](#hessian) |

## HESSIAN

###
|  Version     | Passed | Failed|
|:------------- |:-------:|:-----:|
| 4.0.60  | |:heavy_check_mark:|
| 4.0.51  | |:heavy_check_mark:|
| 4.0.59  | :heavy_check_mark:||
| 4.0.52  | :heavy_check_mark:||
| 4.0.53  | |:heavy_check_mark:|
| 4.0.38  | :heavy_check_mark:||

Here is the new report.

**Comment by wayilau** [comment]
@wu-sheng @IanCao @ascrutae  You may run the test again to see the results now.

**Comment by wayilau** [comment]
@wu-sheng  the error message:
assert failed.
SegmentNotFoundException:
expected:
  Segment:
  - span[0, 1] /HelloService
  - span[-1, 0] /hessian-case/hello

actual:
  Segment[2.28.15622338894210000] e
  expected:     Span[-1, 0] /hessian-case/hello
  actual:       span[-1, 0] /hessian-case/hello
  reason:       [tag(url) value]: expected=>{}, actual=>{http://localhost:18080/hessian-case/hello}

But in expected data expectedata not is {}, it has values, but the reasons tells me that it needs {}. so i was confused.

**Comment by wu-sheng** [comment]
Your yaml format looks like wrong, please recheck.

**Comment by wayilau** [comment]
I have fix the format. @wu-sheng

**Comment by wayilau** [comment]
@wu-sheng  I have found the problem why it fails version together. It just because the port value in the tags is dynamic from SERVER_OUT_PURT.  I have modified it.

**Comment by wayilau** [comment]
# Test Report - 2019-07-05 16:27

- tests  : **6 passed**. **0 failed**
- branch name : **[feature2](https://github.com/apache/incubator-skywalking/tree/feature2)**
- commit id : **[e81507d73a9783dd0af8ebb7d2065c77c790903b](https://github.com/apache/incubator-skywalking/commit/e81507d73a9783dd0af8ebb7d2065c77c790903b)**
- cases branch: **[master](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/tree/master)**
- cases commit id: **[e04db91ad861987bdac3987e120c4b4987835fce](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/commit/e04db91ad861987bdac3987e120c4b4987835fce)**

## Cases List

| Case description | Status | Cases|
|:-----|:-----:|:-----:|
|HESSIAN| **6 passed. 0 failed**| [click me](#hessian) |

## HESSIAN

###
|  Version     | Passed | Failed|
|:------------- |:-------:|:-----:|
| 4.0.60  | :heavy_check_mark:||
| 4.0.51  | :heavy_check_mark:||
| 4.0.59  | :heavy_check_mark:||
| 4.0.52  | :heavy_check_mark:||
| 4.0.53  | :heavy_check_mark:||
| 4.0.38  | :heavy_check_mark:||

**Comment by kezhenxu94** [comment]
@wayilau reopen this when it's ready

**Pull request closed without merging**